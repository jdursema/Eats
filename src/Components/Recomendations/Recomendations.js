import React, { Component } from 'react';
import { fetchRecommendations } from '../../Actions';
import { connect } from 'react-redux';

class Recommendations extends Component {

  async componentDidMount(){
    this.props.handleRecommendations(this.props.favorites, this.props.location, this.props.cuisine)
  }

  render() {
    return (
      <h1>RECOMMENDATIONS</h1>
    );
  }
  
}

const mapStateToProps = state => ({
  favorites: state.favorites,
  location: state.location,
  cuisine: state.cuisine
});

const mapDipatchToProps = dispatch => ({
  handleRecommendations: (favsArray, locationObj, cuisineArray) =>
    dispatch(fetchRecommendations(favsArray, locationObj, cuisineArray))
});

export default connect (mapStateToProps, mapDipatchToProps)(Recommendations)