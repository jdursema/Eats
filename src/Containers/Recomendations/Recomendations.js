/*eslint-disable max-len*/
import React, { Component } from 'react';
import { connect } from 'react-redux';
import CardContainer from '../../Components/CardContainer/CardContainer';
import { postAddFavorite, postDeleteFavorite, fetchRecommendations } from '../../Actions';
import PropTypes from 'prop-types';
import './Recommendations.css'

class Recommendations extends Component {

  async componentDidMount(){
    this.props.handleRecommendations(this.props.favorites, this.props.location, this.props.cuisine);
  }
  render(){
    return (
      <div className='recommendations'>
        <h1>Places you just MIGHT love...</h1>
        <CardContainer
          cards = {this.props.recommendations}
          handleAddFav= {this.props.handleAddFav}
          handleDeleteFav = {this.props.handleDeleteFav}
          user = {this.props.user}
          history = {this.props.history} 
          favorites = {this.props.favorites}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  favorites: state.favorites,
  location: state.location,
  cuisine: state.cuisine,
  recommendations: state.recommendations,
  user: state.user
});

const mapDipatchToProps = dispatch => ({
  handleRecommendations: (favsArray, locationObj, cuisineArray) =>
    dispatch(fetchRecommendations(favsArray, locationObj, cuisineArray)),
  handleAddFav: (cardData, user) => 
    dispatch(postAddFavorite(cardData, user)),
  handleDeleteFav: (cardData, user) => 
    dispatch(postDeleteFavorite(cardData, user))
});

export default connect(mapStateToProps, mapDipatchToProps)(Recommendations);

Recommendations.propTypes = {
  recommendations: PropTypes.array,
  favorites: PropTypes.array,
  handleAddFav: PropTypes.func,
  handleDeleteFav: PropTypes.func,
  user: PropTypes.object,
  history: PropTypes.array,
  handleRecommendations: PropTypes.func,
  cuisine: PropTypes.array,
  location: PropTypes.object
};