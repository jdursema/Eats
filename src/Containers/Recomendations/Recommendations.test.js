import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { Recommendations } from './Recomendations.js';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Recommendations', () => {
  it('should match the snapshot', ()=> {
    const mockFunc = jest.fn();

    const renderedRecommendations = shallow(<Recommendations handleRecommendations= {mockFunc}/>);

    expect(renderedRecommendations).toMatchSnapshot();
  });

  it('should call handleRecommendations when the component mounts', () => {
    const mockFunc = jest.fn();

    shallow(<Recommendations handleRecommendations= {mockFunc}/>);

    expect(mockFunc.mock.calls.length).toEqual(1);
  });
});