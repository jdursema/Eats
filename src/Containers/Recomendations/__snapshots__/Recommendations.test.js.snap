// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommendations should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Recommendations
    handleRecommendations={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Places you just MIGHT love...
</h1>,
        <CardContainer
          cards={undefined}
          favorites={undefined}
          handleAddFav={undefined}
          handleDeleteFav={undefined}
          history={undefined}
          user={undefined}
/>,
      ],
      "className": "recommendations",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Places you just MIGHT love...",
        },
        "ref": null,
        "rendered": "Places you just MIGHT love...",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": undefined,
          "favorites": undefined,
          "handleAddFav": undefined,
          "handleDeleteFav": undefined,
          "history": undefined,
          "user": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Places you just MIGHT love...
</h1>,
          <CardContainer
            cards={undefined}
            favorites={undefined}
            handleAddFav={undefined}
            handleDeleteFav={undefined}
            history={undefined}
            user={undefined}
/>,
        ],
        "className": "recommendations",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Places you just MIGHT love...",
          },
          "ref": null,
          "rendered": "Places you just MIGHT love...",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": undefined,
            "favorites": undefined,
            "handleAddFav": undefined,
            "handleDeleteFav": undefined,
            "history": undefined,
            "user": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
